exercise/submission_models.py

```diff
@@ -28,18 +29,33 @@ def get_queryset(self):
             .prefetch_related('submitters')
 
     def create_from_post(self, exercise, submitters, request):
-        new_submission = Submission.objects.create(
-            exercise=exercise,
-            submission_data=query_dict_to_list_of_tuples(request.POST)
-        )
-        new_submission.submitters.set(submitters)
+
+        submission_data_list = [
+                (key, value) for (key, value) in query_dict_to_list_of_tuples(request.POST)
+                if key != '__aplus__'
+        ]
+        try:
+            meta_data_dict = json.loads(request.POST.get('__aplus__', '{}'))
+        except json.JSONDecodeError:
+            raise ValueError("The content of the field __aplus__ is not valid json")
+        try:
+            new_submission = Submission.objects.create(
+                exercise=exercise,
+                submission_data=submission_data_list,
+                meta_data=meta_data_dict,
+            )
+            new_submission.submitters.set(submitters)
+        except DatabaseError as error:
+            logger.exception("Failed to create submission: %s %s",
+                request.user.username, exercise);
+            raise DatabaseError from error
```

Comment:
I'm not sure if messages got lost due to the long discussion chains, but I tried to say a couple of times that I think `raise DatabaseError from error` could be simply `raise error` or just `raise` here so that it re-raises the same error instance without creating new instances.

Reply:
This is also fine. Benefit is that the top most exception doesn't include the detailed info, which might be beneficial when the exception is rendered to string for API.